cfme/tests/configure/test_ntp_server.py

```diff
@@ -0,0 +1,83 @@
+from cfme import login
+from cfme.configure import configuration
+from cfme.fixtures import pytest_selenium as sel
+from datetime import datetime
+from fixtures.pytest_store import store
+from utils.conf import cfme_data
+from utils.log import logger
+from utils.randomness import generate_random_string
+from utils.ssh import SSHClient
+from utils.wait import wait_for
+
+import pytest
+
+
+def ssh_command_execution(command):
+    with SSHClient() as ssh:
+        status, msg = ssh.run_command("%s" % command)
+        if status == 0:
+            return msg
+
+
+def appliance_date():
+    return datetime.strptime(ssh_command_execution("date --iso-8601=hours").rsplit('-', 1)[0],
+        '%Y-%m-%dT%H')
+
+
+def test_ntp_crud():
+    """ Insert, Update and Delete the NTP servers """
+    # set from yaml file
+    configuration.configure_ntp_servers()
+
+    # set from random values
+    configuration.configure_ntp_servers(*(generate_random_string() for _ in range(3)))
+
+    configuration.unset_ntp_servers()
+
+
+def test_ntp_server_max_character():
+    configuration.configure_ntp_servers(*(generate_random_string(size=255) for _ in range(3)))
+
+
+def test_ntp_conf_file_update_check():
+    configuration.configure_ntp_servers()
+    for clock in cfme_data['clock_servers']:
+        assert ssh_command_execution("cat /etc/ntp.conf | grep %s" % clock).rsplit()
```

Comment:
As this shows in the Error. You should probably make an attempt to make sure msg is not None somewhere. Maybe ssh_command_execution could ```return msg or ''``` so that you always get a string to rsplit on.